Modify auth of password

wangjiaxi90 opened the pull request:
Support password to create vm
Support getSSHConfig with password

wangjiaxi95 committed:
Modify auth of password

Support password to create vm
Support getSSHConfig with password

wangzw commented on builder/qingcloud/config.go:
```diff
@@ -212,6 +213,8 @@ func (config *Config) validate() error {
 		if *describeKeypairOutput.RetCode != 0 || *describeKeypairOutput.TotalCount != 1 {
 			return fmt.Errorf("keypair is not found,%s", *describeKeypairOutput.Message)
 		}
+	} else if len(config.KeypairID) == 0 && config.Password != "" {
```
这里如果 config.KeypairID 和 config.Password 同时为空，报错出来

wangzw commented on builder/qingcloud/builder.go:
```diff
@@ -112,7 +113,17 @@ func (b *Builder) getHost(state multistep.StateBag) (string, error) {
 func (b *Builder) getSSHConfig(state multistep.StateBag) (*gossh.ClientConfig, error) {
 	config := state.Get(BuilderConfig).(Config)
-	privateKey := state.Get(PrivateKey).(string)
+	privateKey, ok := state.Get(PrivateKey).(string)
```
密码和秘钥认证不是互斥的，Auth 是一个数组，如果秘钥和密码同时提供，就都传进去

wangjiaxi95 committed:
Fix bug

1.密码和秘钥认证同时提供，就都传进Auth数组
2.添加对密码强度的校验
3.密码和秘钥可以同时传入state。添加特殊情况：如果密码不为空，秘钥为空，不会自动生成默认秘钥。

wangjiaxi90 committed:
Merge branch 'HashDataInc:master' into master

wangjiaxi95 committed:
Delete check password

1.删除密码校验
